includes/admin/controllers/class-form-fields-admin.php

```diff
@@ -264,8 +269,11 @@ private function hidden_fields_for_type( $field ) {
 	}
 
 	private function delete_field() {
-		// Check permission.
-		check_admin_referer( 'deletefield' );
+		$nonce = wpbdp_get_var( array( 'param' => 'wpbdp_admin_delete_nonce' ), 'post' );
+		if ( ! $nonce ) {
```

Comment:
@theRealRizeo Would you mind moving this nonce check into a different branch so we can get it merged in? Also, I'm not sure why the referer isn't getting checked now if there's no nince.

Reply:
The problem is that the nonce can easily be skipped that way. Is there a better way of checking it when we know we need it checked?

Reply:
I will check, just thought of an easier way to handle this :)